### ISSUE TITLE
Support for Updates with Aggregation Pipeline

### ISSUE THREAD

**Issue by marksto** [congomongo/congomongo#171]
Hi,

There is a limited support for passing [aggregation pipelines](https://docs.mongodb.com/v4.2/core/aggregation-pipeline/) as an `update` argument for update operations. The feature was introduced in MongoDB v4.2. Find the [examples](https://docs.mongodb.com/v4.2/tutorial/update-documents-with-aggregation-pipeline/) in the documentation.

With the latest `congomongo` version `v2.2.3` (which uses Java driver version `3.12.10` that supports the aforementioned feature) an attempt to pass a basic pipeline like this one:
```
(db/fetch-and-modify :collection
                     {:id id}
                     [{:$set {:field "value"}}]
                     :return-new? true
                     :upsert? false)
```
results in the following exception:
```
Execution error (ClassCastException) at somnium.congomongo/fetch-and-modify (congomongo.clj:640).
clojure.lang.LazySeq cannot be cast to com.mongodb.DBObject
```

This feature won't be automagically added in the scope of the ongoing version migration (#164).
It is required to manually extend the logic of the `update!` and `fetch-and-modify` functions.

Cheers,
Mark

**Comment by marksto** [comment]
Ok, so more careful investigation shown up that the Java driver version `3.12.10`, in fact, does not expose that feature.
It's could be used only via `SyncOperations` or `AsyncOperations` internal classes, both of which are marked as "NOT part of the public API".

It happens that only the latest Java driver versions provide this feature.
However, I am not closing this issue so that it can be tracked down.

**Comment by marksto** [comment]
This issue now may be closed whenever we merge #184.
